core/src/banking_stage/unprocessed_transaction_storage.rs

```diff
@@ -177,6 +180,26 @@ fn consume_scan_should_process_packet(
             return ProcessingDecision::Never;
         }
 
+        // Only check fee-payer if we can actually take locks
+        // We do not immediately discard on check lock failures here,
+        // because the priority guard requires that we always take locks
+        // except in the cases of discarding transactions (i.e. `Never`).
+        if payload.account_locks.check_locks(message)
+            && Consumer::check_fee_payer_unlocked(bank, message, &mut payload.error_counters)
+                .is_err()
+        {
+            payload
+                .message_hash_to_transaction
+                .remove(packet.message_hash());
+            return ProcessingDecision::Never;
+        }
+
+        // NOTE:
+        //   This must be the last operation before adding the transaction to the
+        //   sanitized_transactions vector. Otherwise, a transaction could
+        //   be blocked by a transaction that did not take batch locks. This
+        //   will lead to some transactions never being processed, and a
+        //   mismatch in the priorty-queue and hash map sizes.
```

Comment:
nit: not sure if the indent is an editor thing or it's supposed to distinguish from the succeeding comment? probably either remove it or add an empty line as appropriate for clarity

Reply:
b58468d5c8 added a newline. Thanks!